lib/OpenQA/Git/GitServerOutage.pm

```diff
@@ -0,0 +1,39 @@
+
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Git::GitServerOutage;
+use Mojo::Base -strict, -signatures;
+use Mojo::File 'path';
+use Time::Seconds qw(ONE_HOUR);
+use OpenQA::Utils qw(prjdir);
+
+use constant MAX_DURATION_SECONDS => ONE_HOUR / 2;
+
+use constant {
+    SKIP => 'SKIP',
+    FAIL => 'FAIL',
+};
+
+sub _state_file ($app, $server_name) {
+    return path(
+        ($ENV{OPENQA_GIT_SERVER_OUTAGE_FILE} // ($app->config->{global}->{base_dir} // prjdir) . '/git_server_outage')
+        . ".$server_name.lock");
+}
+
+sub remove_state_file ($app, $server_name) {
+    _state_file($app, $server_name)->remove;
+}
+
+sub check_availability ($app, $server_name) {
```

Comment:
This is not really checking the availability. It is meant to report that the server is unavailable and returning whether to fail/skip:
```suggestion
sub report_server_unavailable ($app, $server_name) {
```